src/domain/wallet-invoices/wallet-invoice-builder.ts

```diff
@@ -72,22 +70,45 @@ export const WIBWithRecipient = (state: WIBWithRecipientState): WIBWithRecipient
 }
 
 export const WIBWithExpiration = (state: WIBWithExpirationState): WIBWithExpiration => {
-  const withAmount = async (uncheckedAmount: number) => {
+  const withAmount = async (amount: PaymentAmount<WalletCurrency>) => {
     if (state.recipientWalletDescriptor.currency === WalletCurrency.Usd) {
-      const usdAmount = checkedToUsdPaymentAmount(uncheckedAmount)
+      if (amount.currency === WalletCurrency.Usd) {
+        const usdAmount = amount as UsdPaymentAmount
+        const btcAmount = await state.dealerBtcFromUsd(usdAmount)
+        if (btcAmount instanceof Error) return btcAmount
+        return WIBWithAmount({
+          ...state,
+          hasAmount: true,
+          btcAmount,
+          usdAmount,
+        })
+      }
+
+      const btcAmount = amount as BtcPaymentAmount
+      const usdAmount = await state.dealerUsdFromBtc(btcAmount)
       if (usdAmount instanceof Error) return usdAmount
-      const btcAmount = await state.dealerBtcFromUsd(usdAmount)
-      if (btcAmount instanceof Error) return btcAmount
-
+      if (usdAmount.amount === 0n) {
```

Comment:
We will need to refactor the WalletInvoiceBuilder to be able to switch to the recipient's BTC wallet in this instance, like we did with our payment-flow-builder objects (can be done in a follow-up PR)